dnf/cli/commands/repoquery.py

```diff
@@ -343,55 +349,60 @@ def build_format_fn(self, opts, pkg):
-            t = t.union(query_in.filter(enhances=set_all_deps))
-            t = t.union(query_in.filter(supplements=set_all_deps))
-            t = t.union(query_in.filter(suggests=set_all_deps))
-        if recursive:
-            query_select = t.difference(done)
-            if query_select:
-                done = self._get_recursive_deps_query(query_in, query_select, done=t.union(done),
-                                                      recursive=recursive, all_deps=all_deps)
-        return t.union(done)
+    def _resolve_nevras(self, nevras, base_query):
+        resolved_nevras = self.base.sack.query().filterm(empty=True)
+        for nevra in nevras:
+            resolved_nevras = resolved_nevras.union(base_query.intersection(
+                dnf.subject.Subject(nevra).get_best_query(
+                    self.base.sack,
+                    with_provides=False,
+                    with_filenames=False
+                )
+            ))
+
+        return resolved_nevras
+
+    def _do_recursive_deps(self, query_in, query_select, done=None):
+        done = done if done else query_select
+
+        query_required = query_in.filter(requires=query_select)
+
+        query_select = query_required.difference(done)
+        done = query_required.union(done)
+
+        if query_select:
+            done = self._do_recursive_deps(query_in, query_select, done=done)
 
-    def by_all_deps(self, requires_name, depends_name, query):
-        names = requires_name or depends_name
-        defaultquery = self.base.sack.query().filterm(empty=True)
-        for name in names:
-            defaultquery = defaultquery.union(query.intersection(
-                dnf.subject.Subject(name).get_best_query(self.base.sack, with_provides=False,
-                                                         with_filenames=False)))
-        requiresquery = query.filter(requires__glob=names)
-        if depends_name:
-            requiresquery = requiresquery.union(query.filter(recommends__glob=depends_name))
-            requiresquery = requiresquery.union(query.filter(enhances__glob=depends_name))
-            requiresquery = requiresquery.union(query.filter(supplements__glob=depends_name))
-            requiresquery = requiresquery.union(query.filter(suggests__glob=depends_name))
-
-        done = requiresquery.union(self._get_recursive_deps_query(query, defaultquery,
-                                                                  all_deps=depends_name))
-        if self.opts.recursive:
-            done = done.union(self._get_recursive_deps_query(query, done,
-                                                             recursive=self.opts.recursive,
-                                                             all_deps=depends_name))
         return done
 
+    def by_all_deps(self, names, query, all_dep_types=False):
+        # in case of arguments being NEVRAs, resolve them to packages
+        resolved_nevras = self._resolve_nevras(names, query)
```

Comment:
Using nevra could be confusing because we have NEVRA object. Option parser use REQ for it.

Reply:
What is confusing, the naming? The comment? I'm not sure what you mean...

Reply:
`resolved_nevras` is confusing. Inside the variable there is a query.  What about `nevras_query`, `REQs_query`, `specs_query`, `subjects_query` or anything else that is better than `resolved_nevras`.

Reply:
I've renamed it to `resolved_nevras_query`, as the `_query` suffix seems to be what you really want and while a bit long, it still fits on the lines.